backend/trpc/db.ts

```diff
@@ -0,0 +1,266 @@
+  const CHUNK_SIZE = 900;
+  const allResults: NotebookRow[] = [];
+
+  for (let i = 0; i < accessibleNotebookIds.length; i += CHUNK_SIZE) {
+    const chunk = accessibleNotebookIds.slice(i, i + CHUNK_SIZE);
+    const placeholders = chunk.map(() => "?").join(",");
+    const query = `
+        SELECT id, owner_id, title, created_at, updated_at
+        FROM notebooks
+        WHERE id IN (${placeholders})
+        ORDER BY updated_at DESC
+      `;
+
+    const result = await DB.prepare(query)
+      .bind(...chunk)
+      .all<NotebookRow>();
+
+    allResults.push(...result.results);
+  }
+
+  // Sort all results by updated_at DESC and apply pagination
+  allResults.sort(
+    (a, b) =>
+      new Date(b.updated_at).getTime() - new Date(a.updated_at).getTime()
+  );
+
+  const finalResults = allResults.slice(offset, offset + limit);
+
+  // Add collaborators to each notebook
+  const notebooksWithCollaborators = await Promise.all(
+    finalResults.map(async (notebook) => ({
+      ...notebook,
+      collaborators: await getNotebookCollaborators(DB, notebook.id),
+    }))
+  );
+
+  return notebooksWithCollaborators;
+}
+
+// Helper function to get notebook collaborators
+async function getNotebookCollaborators(db: D1Database, notebookId: string) {
+  const query = `
+    SELECT user_id FROM notebook_permissions
+    WHERE notebook_id = ?
+    AND permission = 'writer'
+  `;
+
+  const writers = await db
+    .prepare(query)
+    .bind(notebookId)
+    .all<Pick<NotebookPermissionRow, "user_id">>();
+
+  if (writers.results.length === 0) {
+    return [];
+  }
+
+  const userIds = writers.results.map((w) => w.user_id);
+  const userMap = await getUsersByIds(db, userIds);
+
+  return userIds.map((userId: string) => {
+    const userRecord = userMap.get(userId);
+    if (userRecord) {
+      return toPublicFacingUser(userRecord);
+    } else {
+      return createFallbackUser(userId);
+    }
+  });
+}
+
+// Get single notebook by ID
+export async function getNotebookById(
+  db: D1Database,
+  notebookId: string
+): Promise<NotebookRow | null> {
+  const notebook = await db
+    .prepare("SELECT * FROM notebooks WHERE id = ?")
+    .bind(notebookId)
+    .first<NotebookRow>();
+
+  return notebook || null;
+}
+
+// Create a new notebook
+export async function createNotebook(
+  db: D1Database,
+  params: {
+    id: string;
+    ownerId: string;
+    title: string;
+    createdAt: string;
+    updatedAt: string;
+  }
+): Promise<boolean> {
+  const { id, ownerId, title, createdAt, updatedAt } = params;
+
+  const result = await db
+    .prepare(
+      `
+      INSERT INTO notebooks (id, owner_id, title, created_at, updated_at)
+      VALUES (?, ?, ?, ?, ?)
+    `
+    )
+    .bind(id, ownerId, title, createdAt, updatedAt)
+    .run();
+
+  return result.success;
+}
+
+// Update notebook metadata
+export async function updateNotebook(
+  db: D1Database,
+  notebookId: string,
+  updates: {
+    title?: string;
+    updatedAt: string;
```

Comment:
Should updatedAt be coming in by call or done at the db layer here? I feel l like we need to account for the server being the main timezone that we convert on the client's end (UTC in the database, converted on the frontend or in response if we know their timestamp).